src/tests/Model/GodunovState.t.h

```diff
@@ -0,0 +1,193 @@
+        {  0.2030313383458647,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,  2.111525918796993e6,   0.0000000000000000,   0.0000000000000000},
+        {  0.2030313383458647,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,  2.111525918796993e6,   0.0000000000000000,   0.0000000000000000},
+        {  0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.6426804463745808,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,  3.341938321147820e6,   0.0000000000000000},
+        {  0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000},
+        {  0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.6426804463745808,   0.0000000000000000,   0.0000000000000000,  3.341938321147820e6},
+        {3.759398496240601e-8,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.3909774436090225,   0.0000000000000000,   0.0000000000000000},
+        {  0.0000000000000000,   0.0000000000000000,   0.0000000000000000, 6.871529877411907e-8,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.3573195536254192,   0.0000000000000000},
+        {  0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000,   0.0000000000000000, 6.871529877411907e-8,   0.0000000000000000,   0.0000000000000000,   0.3573195536254192}
+      }};
+
+      seissol::model::getTransposedGodunovState( local,
+          neighbor,
+          regular,
+          QgodLocal,
+          QgodNeighbor );
+      test_local(QgodLocal, solution_heterogeneous);
+      test_neighbor(QgodNeighbor, solution_heterogeneous);
+
+    }
+
+
+
+    void test_local(init::QgodLocal::view::type QgodLocal, std::array<std::array<real, 9>, 9> solution) {
+      for (int i = 0; i < 9; i++) {
+        for (int j = 0; j < 9; j++) {
+          if (i == j) {
+            test_relative(QgodLocal(i,j), 1-solution[j][i]);
+          }
+          else {
+            test_relative(QgodLocal(i,j), -solution[j][i]);
+          }
+        }
+      } 
+    }
+
+    void test_neighbor(init::QgodLocal::view::type QgodNeighbor, std::array<std::array<real, 9>, 9> solution) {
+      for (int i = 0; i < 9; i++) {
+        for (int j = 0; j < 9; j++) {
+            test_relative(QgodNeighbor(i,j), solution[j][i]);
+        }
+      }
+    }
+
+    void test_nan(init::QgodLocal::view::type QgodNeighbor) {
+      for (int i = 0; i < 9; i++) {
+        for (int j = 0; j < 9; j++) {
+          //CXXTEST TS_ASSERT_IS_NAN is broken
+          assert(std::isnan(QgodNeighbor(i,j)));
```

Comment:
Replace assert with cxxassert, otherwise this would crash the test when it fails